us_election_data_quality_check_and_fixes.sql

```diff
@@ -0,0 +1,206 @@
+from primary_results pr 
+where pr.fips <0 or pr.fips isnull ;
+
+--update tabeli
+update primary_results pr
+set fips = (
+	select cf.fips 
+	from county_facts cf 
+	where cf.area_name = concat(pr.county, ' County') 
+	and cf.state_abbreviation = pr.state_abbreviation)
+where pr.fips isnull ;
+
+--ponowne sprawdzenie, czy fips zawsze dodatni
+select *
+from primary_results pr 
+where fips <0 or fips isnull ;
+
+--sprawdzenie, czy skr�t sk�ada si� zawsze z dw�ch liter
+select 
+	count(*)
+,	case	when upper(state_abbreviation) similar to '[A-Z]{2}' then 'valid state abbr'
+			else 'invalid state abbr' end state_abbr_check
+from primary_results pr 
+group by state_abbr_check ;
+
+--sprawdzenie, czy liczba stan�w odpowiada liczbie skr�t�w
+select 
+	count(distinct state)
+,	count(distinct state_abbreviation)
+from primary_results pr ;
+
+--sprawdzenie, czy skr�t odpowiada nazwie stanu
+select 
+	state_abbreviation 
+,	state 
+from primary_results pr
+group by state_abbreviation, state
+order by state_abbreviation ;
+
+--sprawdzenie, czy stany maj� odpowiedniki w tabeli county_facts 
+select 
+	pr.state_abbreviation as state_abbr_pr
+,	pr.state 
+,	cf.state_abbreviation as state_abbr_cf
+from primary_results pr
+full join county_facts cf on pr.fips = cf.fips 
+group by pr.state_abbreviation, pr.state, cf.state_abbreviation 
+order by pr.state ;
+
+
+--sprawdzenie brakuj�cych odpowiednik�w fips w tabelach county_facts i primary_results
+select  
+	count(distinct pr.fips) as fips_not_in_county_facts
+from primary_results pr 
+full join county_facts cf on pr.fips = cf.fips 
+where cf.fips isnull ;
+--z podzia�em na stany
+select distinct 
+	pr.state_abbreviation 
+,	count(pr.fips) as fips_not_in_county_facts
+from primary_results pr 
+full join county_facts cf on pr.fips = cf.fips 
+where cf.fips isnull 
+group by pr.state_abbreviation ;
+
+--^istnieje 7032 fips�w, niewyst�puj�cych w county_facts
+
+select 
+	count(distinct cf.fips) as fips_not_in_primary_results
+from primary_results pr 
+full join county_facts cf on pr.fips = cf.fips 
+where pr.fips isnull and cf.state_abbreviation is not null ;
+--z podzia�em na stany
+select distinct
+	cf.state_abbreviation 
+,	count(cf.fips) as fips_not_in_primary_results
+from primary_results pr 
+full join county_facts cf on pr.fips = cf.fips 
+where pr.fips isnull and cf.state_abbreviation is not null 
+group by cf.state_abbreviation ;
+
+--^istnieje 335 fips�w niewyst�puj�cych w primary_results 
+--(nieb�d�cych nullami, czyli US lub ca�ymi stanami - te b�d� pozostawione)
+
+--usuni�cie fips�w bez odpowiednik�w
+--poni�sze zapytanie powinno zwraca� 7032 i 335:
+select  
+	count(pr.fips) as fips_not_in_county_facts
+,	count(cf.fips) as fips_not_in_primary_results
+from primary_results pr 
+full join county_facts cf on pr.fips = cf.fips 
+where cf.fips isnull or pr.fips isnull and cf.state_abbreviation is not null ;
+
+delete from primary_results 
```

Comment:
to samo co z updatami - jesli coś deletujemy, warto to umieścić w osobnym skrypcie zbierającym modyfikacje danych